src/callbacks_step/analysis_surface_2d.jl

```diff
@@ -0,0 +1,138 @@
+# By default, Julia/LLVM does not use fused multiply-add operations (FMAs).
+# Since these FMAs can increase the performance of many numerical algorithms,
+# we need to opt-in explicitly.
+# See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
+@muladd begin
+#! format: noindent
+# TODO - Add explanation for what this file contains
+
+struct AnalysisSurfaceIntegral{Semidiscretization, Indices, Variable}
+    semi::Semidiscretization
+    indices::Indices
+    variable::Variable # Quantity of interest, like lift or drag
+    function AnalysisSurfaceIntegral(semi, boundary_condition_type, variable)
+        # The bc list as ordered in digest_boundary_conditions
+        ordered_bc = semi.boundary_conditions.boundary_condition_types
+
+        # The set of all indices that gives the bc where the surface integral is to be computed
+        index = sort(findall(x -> x == boundary_condition_type, ordered_bc))
```

Comment:
I think it would be better to grab the indices based on the boundary name, e.g., `:Airfoil` instead of the `boundary_condition_type`. I am thinking of situations like if one runs the channel flow over a cylinder test case. There the cylinder as well as the top and bottom of the domain are set as a wall boundary condition. So if one used the current implementation to attempt and compute the drag on the cylinder there would be extra integral contributions of the drag from the top and bottom walls.

Reply:
This approach may require us to make a change in mesh file (unless I misunderstand something), which the user may not always have control over.

However, with the current approach, only elixir file needs to be changed. The way to handle the case you specified would be something like
```julia
boundary_condition_slip_wall_cylinder(args...) = boundary_condition_slip_wall(args...)
```
and pass `boundary_condition_slip_wall_cylinder` as the boundary conditions only for the boundary names over which you want to integrate.

Reply:
> The approach you suggest may require us to make a change in mesh file

Possibly yes. The approach I suggested assumes that the user set appropriate boundary names in the mesh generator, which might not be the case, e.g., if the mesh was "inherited" by the user from elsewhere. Creating a small wrapper function to ensure the correct drag forces (for instance) are computed could work.

> Also, another issue with the approach you suggested is that I am not sure if the `digest_boundary_conditions` function keeps the boundary names in the `Semidiscretization`.

That is a good point that I forgot to check. The boundary names are only saved in the `BoundaryContainer` of the unstructured meshes. So you are right, and the callback would need to have additional information in the struct (possibly).

I am open to either strategy, but it would be good to get input from others, e.g., @DanielDoehring @sloede @ranocha

Reply:
If we go in the direction of using boundary names, here is a way I would suggest -

1) User passes the boundary names they want the integral to be computed on, along with the `boundary_conditions` dictionary
2) Internally, the `boundary_conditions` dictionary is used to map those boundary names to boundary conditions. We will put as assertion to ensure that those boundary conditions are not values for any other keys in the dictionary. (Thus there is still some work required by the user, but there cannot be any mistakes anymore)
3) Now we use the approach that is currently in the PR, of using the boundary conditions to obtain the indices where the integral is to be computed.

Reply:
I agree with @andrewwinters5000 that specifying the relevant boundaries should be done via name, not type.

> I am not sure if the `digest_boundary_conditions` function keeps the boundary names in the Semidiscretization

I can take a look at this in some time since I have relatively recently checked that functionality when coding up the BC association based on nodesets.

Reply:
👍

Reply:
Okay, I finally had the time to look at this again.

In principle, one could pass in the Symbol of the boundary for which forces should be computed.

Based on this symbol, one can query the `boundary_condition` dict for this symbol to obtain the corresponding BC type, which leads to the way how it is currently implemented.

In order to be able to compute forces for different boundaries that have the same condition we need an additional datastructure `boundary_indices_per_symbol` or similar that can then be queried for a certain symbol. I will look into this.

Reply:
I just pushed this approach. I checked that results agree with the previous implementation for the Euler airfoil